src/engraving/libmscore/slur.cpp

```diff
@@ -416,9 +416,29 @@ void SlurSegment::avoidCollisions(PointF& pp1, PointF& p2, PointF& p3, PointF& p
 
     // Collision clearance at the center of the slur
     double clearance = 0.75 * spatium(); // TODO: style
-    // balance: determines how much endpoint adjustment VS shape adjustment
-    // we will do (0 = only shape, 1 = only end point)
-    double balance = 0.4; // TODO: style (maybe?)
+    // balance: determines how much endpoint adjustment VS shape adjustment we will do.
+    // 0 = end point is fixed, only the shape can be adjusted,
+    // 1 = shape is fixed, only end the point can be adjusted.
+    // left and right side of the slur may have different balance depending on context:
+    double leftBalance, rightBalance;
+    if (isSingleBeginType()) {
+        if (startCR->isChord() && toChord(startCR)->stem() && startCR->up() == slur()->up()) {
```

Comment:
Hm, actually, I realize now that it's not yet 100% guaranteed at this point that `startCR != nullptr` (same for `endCR`). Either add a check `if (!startCR || !endCR) { return; }` at the top of the function so that you don't need to worry about it anymore, or add a null check here (`if (startCR && startCR->isChord() && ...`)

Reply:
Ha! good catch. Yeah, if either of them is null it just makes sense to return.

Reply:
Yup, just did that :)